### ISSUE TITLE
Inconsistent output between bpftool and cilium map get for cilium_lb4_backends_v2 and many duplicated entries with different id in this map.

### ISSUE THREAD

**Issue by ghost** [cilium/cilium#23551]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

Failed to access coredns with service ip,  but accessing with pod ip is working.

The log of cilium agent shows like:
```
level=error msg="Error while inserting service in LB map" error="Unable to add backend .... 
    Unable to update element for map cilium_lb4_backends_v2 map with the file descriptor 13: 
    the map is full, plesase consider resizing it. argument list too long" ...
```
which indicates maybe the cilium_lb4_backends_v2 is full

Output of `cilium map list | grep backends`
```
cilium_lb4_backends_v2    14052   2782     true
``` 
means that we have 14052 enties and 2782 errors. Considering the max entries for lb map is 65536, this should have some space for more data.

However, when using bpftool, the number of items are 65536 already

`bpftool map dump pinned /sys/fs/bpf/tc/globals/cilium_lb4_backends_v2 | tail -n 1`

```
Found 65535 elements
```

Besides, there are many duplicate values (ip and port) in this map, while only one of them is in the output of `cilium map get  cilium_lb4_backends_v2`. I guess it should be that only one id for one unique value.   
I am confused that why `cilium map get` cannot get all the entries in this map, and why there are many duplicated entries.

### Cilium Version

1.11.4

### Kernel Version

5.10.0-60.61.0.88.oe2202.x86_64

### Kubernetes Version

1.24.3

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by brb** [comment]
Thanks for the issue.

> Output of cilium map list | grep backends
>> cilium_lb4_backends_v2    14052   2782     true

Please use `cilium bpf lb list --backends` instead.

**Comment by nijich** [comment]
> `cilium bpf lb list --backends` should give you a consistent view.

Hi, the result of this command is `Error: unknown flag: --backends`

The version of cilium is 1.11.4.

while `cilium lb list | wc -l ` gives a result: 28159.

I still cannot get it but I will search for the different between cilium bpf and cilium map.

**Comment by nijich** [comment]
@brb Sorry for bothering you again. After watching code, I still don't understands why there are many duplicated backends. For example,  using `cilium bpf lb list --backends | awk '{print $2}' | sort | uniq -c` shows that many backends occurs for 1-8 times. And I use `cilium bpf lb list --frontends |grep <backend-id>` one by one for one of the duplicated backends, only one id match a result.  

Could you give me some instructions to understand this? I want to know that how to use the size of this map to calculate the max service number for a k8s cluster. Thank you very much.

**Comment by hemanthmalla** [comment]
We're noticing the same behavior in some nodes after upgrading from 1.10 to 1.11
Curious if this has something to do with u16 to u32 backend ID change in 1.11

@nijich did you recently upgrade as well ?
Inconsistent state went away for us after an agent restart.

@brb any idea what can cause duplicate backend entries ? Thanks

**Comment by nijich** [comment]
@hemanthmalla We don't have any upgrade. And we don't restart the pod since we want to find the root cause.

**Comment by aditighag** [comment]
@nijich Hey, so we fixed the leak in - https://github.com/cilium/cilium/pull/23858, and the PR that deletes leaked backends was backported to 1.12 and 1.13 branches - https://github.com/cilium/cilium/pull/24681. The 24681 fix should be released in 1.12.9 and 1.13.2 soon. It'll be great if you can try that out. Thanks! If you are planning to test out the fix, we can leave this issue open for a bit.

**Comment by nijich** [comment]
@aditighag thanks. I tried v1.11.15 and the leak problem seems to be fixed. Previous leaked enties was delete by bpftool manually. I think this issue can be closed.
